astropy/io/ascii/ui.py

```diff
@@ -660,9 +660,27 @@ def write(table, output=None,  format=None, Writer=None, fast_writer=True, **kwa
         List of names to exclude from output (applied after ``include_names``)
     fast_writer : bool
         Whether to use the fast Cython writer (default= ``True``)
+    overwrite : bool
+        Whether to overwrite an existing file (default= ``False``).
+        This parameter is ignored when the output is not a string
+        (e.g., a file object).
+        Since the default behaviour is changed (previously, files would
+        always be overwritten), an AstropyDeprecationWarning is issued
+        when this parameter is not specified and the file already exists.
     Writer : ``Writer``
         Writer class (DEPRECATED) (default= :class:`Basic`)
     """
+    overwrite = kwargs.pop('overwrite', None)
+    if isinstance(output, six.string_types):
+        if os.path.isfile(output):
```

Comment:
If `output` is a directory then this gives `False`.  It will fail later trying to overwrite a directory, but I assume we would prefer to catch the error earlier if `overwrite is not True`.  `os.path.isfile` also gives `False` for an unresolved symlink.  Again I would think that we don't want to overwrite that if `overwrite is not True`.  So maybe `os.path.lexists()`?